packages/search/addon/services/navi-search-result.js

```diff
@@ -0,0 +1,26 @@
+/**
+ * Copyright 2020, Yahoo Holdings Inc.
+ * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
+ *
+ * This service is used to discover all the available search providers.
+ */
+
+import Service from '@ember/service';
+import config from 'ember-get-config';
+
+/* global requirejs */
+
+export default class NaviSearchResultService extends Service {
+  /**
+   * @method all
+   * @returns {Array} array of available search providers
+   */
+  all() {
+    const resultComponentRegex = new RegExp(`^(?:${config.modulePrefix}/)?components/navi-search-result/([a-z-]*)$`),
+      auxResultComponentRegex = new RegExp(`^(?:${config.modulePrefix}/)?components/([a-z-/]*)$`),
```

Comment:
What is this for? this wasn't part of the rfc.

Reply:
this regex matches the name of the component together with the folder so for example from `dummy/components/navi-search-result/report` it will return `navi-search-result/report`

Reply:
Will add that to the RFC as well.